src/Console/Output/Progress/ProgressOutputFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output\Progress;
+
+use PhpCsFixer\Console\Output\OutputContext;
+
+/**
+ * @internal
+ */
+final class ProgressOutputFactory
+{
+    public function create(string $outputType, OutputContext $context): ProgressOutputInterface
+    {
+        if (null === $context->getOutput()) {
+            $outputType = ProgressOutputType::NONE;
+        }
+
+        if (!self::isBuiltInType($outputType)) {
+            throw new \RuntimeException(
```

Comment:
`InvalidArgumentException`  ?

Reply:
I'll also change `self::isBuiltInType()` to `$this->isBuiltInType()` because method is not static (it was at some point I believe, but it was refactored several times).